enclave/src/test_proxy.rs

```diff
@@ -0,0 +1,128 @@
+pub fn test_check_main_account() {
+    pointer_initialize();
+    let account_to_find_real: [u8; 32] = Vec::from("first_account").using_encoded(blake2_256);
+    let account_to_find_false: [u8; 32] = Vec::from("false_account").using_encoded(blake2_256);
+    assert_eq!(polkadex::check_main_account(account_to_find_real), Ok(true));
+    assert_eq!(
+        polkadex::check_main_account(account_to_find_false),
+        Ok(false)
+    );
+}
+
+#[allow(unused)]
+pub fn test_check_proxy_account() {
+    pointer_initialize();
+    let main_account: [u8; 32] = Vec::from("first_account").using_encoded(blake2_256);
+    let main_account_false: [u8; 32] = Vec::from("false_account").using_encoded(blake2_256);
+    let dummy_account_one: [u8; 32] = Vec::from("first_dummy_account").using_encoded(blake2_256);
+    let dummy_account_false: [u8; 32] = Vec::from("false_dummy_account").using_encoded(blake2_256);
+    assert_eq!(
+        polkadex::check_proxy_account(main_account, dummy_account_one),
+        Ok(true)
+    );
+    assert_eq!(
+        polkadex::check_proxy_account(main_account, dummy_account_false),
+        Ok(false)
+    );
+    assert_eq!(
+        polkadex::check_proxy_account(main_account_false, dummy_account_false),
+        Ok(false)
+    );
+}
+
+#[allow(unsued)]
+pub fn test_add_main_account() {
+    pointer_initialize();
+    let main_account: [u8; 32] = Vec::from("new_account").using_encoded(blake2_256);
+    polkadex::add_main_account(main_account);
+    let mutex = load_proxy_registry().unwrap();
+    let mut proxy_storage = mutex.lock().unwrap();
+    assert_eq!(proxy_storage.accounts.contains_key(&main_account), true);
+}
+
+#[allow(unsued)]
+pub fn test_remove_main_account() {
+    pointer_initialize();
+    let main_account: [u8; 32] = Vec::from("first_account").using_encoded(blake2_256);
+    polkadex::remove_main_account(main_account);
+    let mutex = load_proxy_registry().unwrap();
+    let mut proxy_storage = mutex.lock().unwrap();
+    assert_eq!(proxy_storage.accounts.contains_key(&main_account), false);
+}
+
+#[allow(unsued)]
+pub fn test_add_proxy_account() {
+    pointer_initialize();
+    let main_account: [u8; 32] = Vec::from("first_account").using_encoded(blake2_256);
+    let new_proxy_account: [u8; 32] = Vec::from("new_account").using_encoded(blake2_256);
+    polkadex::add_proxy(main_account, new_proxy_account);
+    let mutex = load_proxy_registry().unwrap();
+    let mut proxy_storage = mutex.lock().unwrap();
+    let proxies = proxy_storage.accounts.get(&main_account).unwrap();
+    assert_eq!(proxies.contains(&new_proxy_account), true);
+}
+
+#[allow(unsued)]
+pub fn test_remove_proxy_account() {
+    pointer_initialize();
+    let main_account: [u8; 32] = Vec::from("first_account")
+        .encode()
+        .using_encoded(blake2_256);
+    let dummy_account_one: [u8; 32] = Vec::from("first_dummy_account").using_encoded(blake2_256);
+    polkadex::remove_proxy(main_account, dummy_account_one);
```

Comment:
Same as above: could you ensure here that dummy_account_one acutally existed before removal?

Reply:
resolved